evalml/automl/automl_search.py

```diff
@@ -450,7 +450,7 @@ def _get_batch_number(self):
 
     def _pre_evaluation_callback(self, pipeline):
         if self.start_iteration_callback:
-            self.start_iteration_callback(pipeline.__class__, pipeline.parameters, self)
+            self.start_iteration_callback(pipeline, pipeline.parameters, self)
```

Comment:
While we're making a breaking change here: I think we should delete the `parameters` argument.

Why: now that we're providing a pipeline instance, the start iteration callback can access the parameters by calling `pipeline.parameters`, so there's no need for that to also be provided as a standalone arg.

Reply:
Done!